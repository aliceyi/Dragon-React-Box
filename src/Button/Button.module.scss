@import '@/styles/base.scss';

.ff-button {
  &__wrapper {
    display: flex;
    justify-content: center;

    &.bottom {
      width: 100%;
      margin: 0;
      padding: 0 $size-m;
      position: fixed;
      bottom: 0;
      bottom: constant(safe-area-inset-bottom);
      bottom: env(safe-area-inset-bottom);
      left: 0;
      padding-bottom: $size-s;
    }

    &__button {
      position: relative;
      padding: 0;
      // border-radius: 4px;
      border: $border-btn-dark;
      height: 42px;
      font-family: $font-family-semibold;
      font-size: $font-size-15;
      line-height: 42px;
      color: $color-white;
      background-color: $color-dark;
      outline: none;
      overflow: hidden;
      cursor: pointer;

      &--white {
        color: $color-dark;
        background-color: $color-white;
        border: $border-btn-dark;
      }

      &--disabled {
        color: $color-white;
        background-color: $color-grey;
        border: $color-grey;
      }

      &--noBorder {
        border: none;
      }

      &__icon {
        position: absolute;
        line-height: 19px;
        right: $size-s;
        top: $size-s;
      }
    }

    &__a {
      color: currentColor;
      font-family: $font-family-semibold;
      font-size: $font-size-15;
      line-height: 42px;
      text-decoration: none;
      width: 100%;
      text-align: center;

      &:link,
      &:visited,
      &:hover,
      &:active {
        color: currentColor;
        font-family: $font-family-semibold;
        font-size: $font-size-15;
        line-height: 42px;
        text-decoration: none;
      }
    }
  }
}