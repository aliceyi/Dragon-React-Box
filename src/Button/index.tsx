import React from 'react'
import { IconPlus } from '../SvgIcons'
import { MouseEvent } from 'react'
import styles from './Button.module.scss'

type Props = {
  width?: string
  fixed?: string
  white?: boolean
  icon?: string
  link?: string
  noBorder?: boolean
  classNames?: string
  type?: 'submit' | 'reset' | 'button'
  disabled?: boolean
  onClick?: (e: MouseEvent<HTMLButtonElement>) => void
}

const Button: React.FunctionComponent<Props> = ({ children, width, fixed, white, icon, link, noBorder, classNames, type, disabled = false, onClick }) => {
  const wrapperClasses = [styles['ff-button__wrapper']]
  fixed && wrapperClasses.push(styles[fixed])
  if (classNames) {
    wrapperClasses.push(classNames)
  }
  const btnClasses = [styles['ff-button__wrapper__button']]
  white && btnClasses.push(styles['ff-button__wrapper__button--white'])
  disabled && btnClasses.push(styles['ff-button__wrapper__button--disabled'])
  noBorder && btnClasses.push(styles['ff-button__wrapper__button--noBorder'])

  const handleClick = (e: any) => {
    if (type !== 'reset' && type !== 'submit') { e.preventDefault() }
    !disabled && onClick && onClick(e)
  }

  return (
    <div className={wrapperClasses.join(' ')}>
      {!link ?
        <button className={btnClasses.join(' ')} style={{ width: width || '100%' }} type={type} onClick={handleClick}>
          <span>{children}</span>
          {icon && icon === 'plus' ? <span className={styles['ff-button__wrapper__button__icon']}><IconPlus width="19px" height="19px" /></span> : null}
        </button> :
          <a href={link} className={styles['ff-button__wrapper__a']}>{children}</a>
        }
    </div>
  )
}

export default Button
