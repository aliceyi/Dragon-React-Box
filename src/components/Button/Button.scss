@use '../../styles/base' as *;

.ff__button {
  //   &__wrapper {
  // display: flex;
  // justify-content: center;

  & {
    display: flex;
    justify-content: center;
    position: relative;
    padding: 0 $size-s;
    // border-radius: 4px;
    height: $btn-line-height;
    font-family: $btn-font-family;
    font-size: $btn-font-size;
    line-height: $btn-line-height;
    outline: none;
    overflow: hidden;
    cursor: pointer;

    @include button-style($btn-font-color-dark, $btn-background-color-white, $btn-border-dark);

    &--default {
      @include button-style($btn-font-color-dark, $btn-background-color-white, $btn-border-dark);
    }

    &--primary {
      @include button-style($btn-font-color-white, $btn-background-color-dark, $btn-border-dark);
    }

    &--disabled {
      @include button-style($btn-font-color-white, $btn-background-color-disabled, $btn-border-disabled);
      color: $btn-font-color-white;
      background-color: $btn-background-color-disabled;
      border: $btn-border-disabled;
      cursor: not-allowed;

      >* {
        pointer-events: none;
      }
    }

    &--noBorder {
      border: none;
    }

    &--large {
      @include height($btn-large-line-height);
    }

    &--medium {
      @include height($btn-line-height);
    }

    &--small {
      @include height($btn-small-line-height);
    }

    &--text-left {
      display: flex;

      .content {
        text-align: left;
        flex: 1;
      }
    }

    &--text-center {
      display: flex;

      .content {
        text-align: center;
        flex: 1;
      }
    }

    &--text-right {
      display: flex;

      .content {
        text-align: right;
        flex: 1;
      }
    }


    &__icon {
      position: absolute;
      line-height: 19px;
      right: $size-s;
      top: $size-s;
    }
  }

  &--link {
    background: none;
    border: none;
    color: $btn-link-color;
    font-family: $btn-font-family;
    font-size: $btn-font-size;
    line-height: $btn-line-height;
    text-decoration: $link-decoration;
    width: 100%;
    text-align: center;
    outline-style: none;

    &:visited,
    &:hover,
    &:active {
      color: $btn-link-hover-color;
      font-family: $font-family-semibold;
      font-size: $font-size-15;
      line-height: $btn-line-height;
      text-decoration: $link-hover-decoration;
    }

    &:disabled,
    &.disabled {
      color: $btn-link-disabled-color;
      pointer-events: $link-hover-decoration;
    }
  }

  //   }
}
