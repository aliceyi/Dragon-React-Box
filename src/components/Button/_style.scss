// @import '@/styles/base.scss';

.ff-button {
//   &__wrapper {
    // display: flex;
    // justify-content: center;

    &__button {
      display: flex;
      justify-content: center;
      position: relative;
      padding: 0;
      // border-radius: 4px;
      height: $btn-line-height;
      font-family: $btn-font-family;
      font-size: $btn-font-size;
      line-height: $btn-line-height;
      outline: none;
      overflow: hidden;
      cursor: pointer;

      @include button-style($btn-font-color-dark, $btn-background-color-white, $btn-border-dark);

      &--default {
        @include button-style($btn-font-color-dark, $btn-background-color-white, $btn-border-dark);
      }

      &--primary {
        @include button-style($btn-font-color-white, $btn-background-color-dark, $btn-border-dark);
      }

      &--disabled {
        @include button-style($btn-font-color-white, $btn-background-color-disabled, $btn-border-disabled);
        color: $btn-font-color-white;
        background-color: $btn-background-color-disabled;
        border: $btn-border-disabled;
        cursor: not-allowed;
        > * {
          pointer-events: none;
        }
      }

      &--noBorder {
        border: none;
      }

      &__icon {
        position: absolute;
        line-height: 19px;
        right: $size-s;
        top: $size-s;
      }
    }

    &__button--link {
      background: none;
      border: none;
      color: $btn-link-color;
      font-family: $btn-font-family;
      font-size: $btn-font-size;
      line-height: $btn-line-height;
      text-decoration: $link-decoration;
      width: 100%;
      text-align: center;

      &:visited,
      &:hover,
      &:active {
        color: $btn-link-hover-color;
        font-family: $font-family-semibold;
        font-size: $font-size-15;
        line-height: $btn-line-height;
        text-decoration: $link-hover-decoration;
      }
      &:disabled,
      &.disabled {
        color: $btn-link-disabled-color;
        pointer-events: $link-hover-decoration;
      }
    }
//   }
}
