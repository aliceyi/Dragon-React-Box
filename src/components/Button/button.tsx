import React, { FC, ButtonHTMLAttributes, AnchorHTMLAttributes } from 'react';
// import './button.css';
import classNames from 'classnames'

export interface BaseButtonProps {
    /**
     * setting button style
     */
    className?: string;
    /**
     * Is this the principal call to action on the page?
     */
    btnType?: 'primary' | 'default' | 'danger' | 'link' | 'noBorder';
    /**
     * What background color to use
     */
    backgroundColor?: string;
    /**
     * How large should the button be?
     */
    size?: 'small' | 'medium' | 'large';
    /**
     * button children node
     */
    children: React.ReactNode;
    /**
     * when btnType is a link, href have to value
     */
    href?: string;
    /**
     * setting test id
     */
    testData?: string;
    /**
     * setting disabled
     */
    disabled?: boolean;
}

type NativeButtonProps = BaseButtonProps & ButtonHTMLAttributes<HTMLElement>
type AnchorButtonProps = BaseButtonProps & AnchorHTMLAttributes<HTMLElement>
export type ButtonProps = Partial<NativeButtonProps & AnchorButtonProps>

/**
 * Primary UI component for user interaction
 */
export const Button: FC<ButtonProps> = ({
    btnType = 'default',
    children,
    size = 'medium',
    href,
    testData,
    className,
    disabled = false,
    ...props
}) => {
    // const mode = btnType ? 'storybook-button--primary' : 'storybook-button--secondary';
    const btnClasses = classNames(
        'ff__button',
        className,
        {
            [`ff__button--${btnType}`]: btnType,
            [`ff__button--disabled`]: disabled,
        })
    if (btnType === 'link' && href) {
        return (
            <a
                href={href}
                className={btnClasses}
                data-testid={testData}
                {...props}
            >{children}</a>
        )
    } else {
        return (
            <button
                className={btnClasses}
                data-testid={testData}
                disabled={disabled}
                {...props}
            >
                <span>{children}</span>
                {/* {icon && icon === 'plus' ? <span className="ff__button__icon"><IconPlus width="19px" height="19px" /></span> : null} */}
            </button>
        )
    }
};


export default Button;