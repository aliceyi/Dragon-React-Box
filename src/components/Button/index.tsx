import React, { FC, ButtonHTMLAttributes, AnchorHTMLAttributes } from 'react';
import './Button.scss';
import classNames from 'classnames'

export interface BaseButtonProps {
    /**
     * setting button style
     */
    className?: string;
    /**
     * Is this the principal call to action on the page?
     */
    btnType?: 'primary' | 'default' | 'link';
    /**
     * How large should the button be?
     */
    size?: 'small' | 'medium' | 'large';
    /**
     * button children node
     */
    children: React.ReactNode;
    /**
     * when btnType is a link, href have to value
     */
    href?: string;
    /**
     * setting test id
     */
    testData?: string;
    /**
     * setting disabled
     */
    disabled?: boolean;
    /**
     * setting button  text alignment
     */
    textAlignment?: 'left' | 'center' | 'right';
    /**
     * setting button  icon alignment
     */
    iconAlignment?: 'left' | 'right';
    /**
     * setting button width
     */
    width?: string;
}

type NativeButtonProps = BaseButtonProps & ButtonHTMLAttributes<HTMLElement>
type AnchorButtonProps = BaseButtonProps & AnchorHTMLAttributes<HTMLElement>
export type ButtonProps = Partial<NativeButtonProps & AnchorButtonProps>

/**
 * Primary UI component for user interaction
 */
export const Button: FC<ButtonProps> = ({
    btnType = 'default',
    children,
    size = 'medium',
    href,
    testData,
    className,
    disabled = false,
    textAlignment = 'center',
    iconAlignment = 'left',
    width = '180px',
    ...props
}) => {
    // const mode = btnType ? 'storybook-button--primary' : 'storybook-button--secondary';
    const btnClasses = classNames(
        'ff__button',
        className,
        {
            [`ff__button--${btnType}`]: btnType,
            [`ff__button--disabled`]: disabled,
            [`ff__button--${size}`]: size,
            [`ff__button--text-${textAlignment}`]: textAlignment,
            [`ff__button--icon-${iconAlignment}`]: iconAlignment
        })
    if (btnType === 'link' && href) {
        return (
            <a
                style={{ width: width }}
                href={href}
                className={btnClasses}
                data-testid={testData}
                {...props}
            >{children}</a>
        )
    } else {
        return (
            <button
                style={{ width: width }}
                className={btnClasses}
                data-testid={testData}
                disabled={disabled}
                {...props}
            >
                <span className="content">{children}</span>
                {/* {icon && icon === 'plus' ? <span className="ff__button__icon"><IconPlus width="19px" height="19px" /></span> : null} */}
            </button>
        )
    }
};


export default Button;
