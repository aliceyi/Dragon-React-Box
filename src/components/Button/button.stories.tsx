import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

// import { action } from '@storybook/addon-actions';
import { Button, ButtonProps } from './button';

export default {
  title: 'Example/MyButton',
  component: Button,
  argTypes: {
    onClick: { action: 'mybutton clicked' },
    backgroundColor: { control: 'color' },
    btnType: {
      control: { type: 'select', options: ['primary', 'default', 'white'] },
    },
  }
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  btnType: 'primary',
  children: 'primary btn'
  // width: '100px',
  // onClick: action('my button click')
};

export const Link = Template.bind({});
Link.args = {
  btnType: 'link',
  children: 'link btn'
}


// import React from 'react';
// // also exported from '@storybook/react' if you can deal with breaking changes in 6.1
// import { Story, Meta } from '@storybook/react/types-6-0';

// import { Button, ButtonProps } from './button';

// export default {
//   title: 'Example/Button12',
//   component: Button,
//   argTypes: {
//     backgroundColor: { control: 'color' },
//   },
// } as Meta;

// const Template: Story<ButtonProps> = (args) => <Button {...args} />;

// export const Primary = Template.bind({});
// Primary.args = {
//   btnType: 'default',
//   // label: 'Button',
// };

// export const Secondary = Template.bind({});
// Secondary.args = {
//   label: 'Button',
// };

// export const Large = Template.bind({});
// Large.args = {
//   size: 'large',
//   label: 'Button',
// };

// export const Small = Template.bind({});
// Small.args = {
//   size: 'small',
//   label: 'Button',
// };
