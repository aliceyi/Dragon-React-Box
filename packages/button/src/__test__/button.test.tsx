import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import Button, { ButtonProps } from '../index'
// https://github.com/styled-components/jest-styled-components
import 'jest-styled-components'

const defaultProps = {
    onClick: jest.fn(),
    testData: 'default',
}

const testProps: ButtonProps = {
    width: '100px',
    className: 'testBtn',
    btnType: 'button',
    testData: 'testProps',
}

const linkProps: ButtonProps = {
    btnType: 'text',
    testData: 'text',
    href: 'http://www.farfetch.com',
}

const disableProps: ButtonProps = {
    testData: 'disabled',
    disabled: true,
    onClick: jest.fn(),
}

describe('test Button component', () => {
    it('should render the correct default button', () => {
        render(<Button {...defaultProps}>Default</Button>)
        const element = screen.getByTestId('default')
        expect(element).toBeInTheDocument()
        expect(element.tagName).toEqual('BUTTON')
        // expect(element).toHaveClass('ff__button')
        // expect(element.disabled).toBeFalsy()
        fireEvent.click(element)
        expect(defaultProps.onClick).toHaveBeenCalled()
        expect(element).toMatchSnapshot()
    })
    it('should render the correct component based on different props', () => {
        render(<Button {...testProps}>test Props</Button>)
        const element = screen.getByTestId('testProps')
        expect(element).toBeInTheDocument()
        // expect(element).toHaveClass('ff__button ff__button--primary testBtn')
        expect(element).toMatchSnapshot()
    })
    it('should render a link when btnType equals link and href is provided', () => {
        render(<Button {...linkProps}>Link</Button>)
        const element = screen.getByTestId('text')
        expect(element).toBeInTheDocument()
        expect(element.tagName).toEqual('A')
        // expect(element).toHaveClass('ff__button ff__button--link')
        expect(element).toMatchSnapshot()
    })
    it('should render disabled button when disabled set to true', () => {
        render(<Button {...disableProps}>Disabled</Button>)
        const element = screen.getByTestId('disabled')
        expect(element).toBeInTheDocument()
        // expect(element.disabled).toBeTruthy()
        fireEvent.click(element)
        expect(disableProps.onClick).not.toHaveBeenCalled()
        expect(element).toMatchSnapshot()
    })
})
