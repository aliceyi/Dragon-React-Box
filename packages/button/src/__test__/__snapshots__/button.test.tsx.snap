// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component should render a link when btnType equals link and href is provided 1`] = `
.c0 {
  color: #222222;
  background-color: transparent;
  border: 1px solid transparent;
  width: 180px;
  display: inline-block;
}

.c0:hover {
  color: #222222;
  background: 141414;
  border-color: a0a0a;
}

.c0:focus,
.c0.focus {
  color: #222222;
  background: 141414;
  border-color: a0a0a;
}

.c0:disabled,
.c0.disabled {
  color: #222222;
  background: transparent;
  border-color: transparent;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > svg {
  float: left;
  width: 20px;
  height: 20px;
  margin-right: 6px;
}

.c1 > span {
  vertical-align: middle;
  white-space: nowrap;
  line-height: 42px;
  font-size: 0.9375rem;
}

<a
  class="c0"
  data-testid="text"
  width="180px"
>
  <div
    class="c1"
  >
    <span>
      Link
    </span>
  </div>
</a>
`;

exports[`test Button component should render disabled button when disabled set to true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  padding: 0 12px;
  height: 42px;
  font-family: 'PingFangSC-Light','sans-serif-light','Helvetica Neue For Number','-apple-system',BlinkMacSystemFont,'Segoe UI',Roboto,'Hiragino Sans GB','Microsoft YaHei','Helvetica Neue',Helvetica,Arial;
  font-size: 0.9375rem;
  line-height: 42px;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  width: 180px;
  color: #ffffff;
  background-color: #bfbfbf;
  border: 1px solid #bfbfbf;
}

.c0:hover {
  color: #ffffff;
  background: #d3d3d3;
  border-color: #c9c9c9;
}

.c0:focus,
.c0.focus {
  color: #ffffff;
  background: #d3d3d3;
  border-color: #c9c9c9;
}

.c0:disabled,
.c0.disabled {
  color: #ffffff;
  background: #bfbfbf;
  border-color: #bfbfbf;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > svg {
  float: left;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 > span {
  vertical-align: middle;
  white-space: nowrap;
}

<button
  class="c0"
  data-testid="disabled"
  disabled=""
  type="button"
  width="180px"
>
  <div
    class="c1"
  >
    <span>
      Disabled
    </span>
  </div>
</button>
`;

exports[`test Button component should render the correct component based on different props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  padding: 0 12px;
  height: 42px;
  font-family: 'PingFangSC-Light','sans-serif-light','Helvetica Neue For Number','-apple-system',BlinkMacSystemFont,'Segoe UI',Roboto,'Hiragino Sans GB','Microsoft YaHei','Helvetica Neue',Helvetica,Arial;
  font-size: 0.9375rem;
  line-height: 42px;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  width: 100px;
  color: #222222;
  background-color: #ffffff;
  border: 1px solid #222222;
}

.c0:hover {
  color: #222222;
  background: #ffffff;
  border-color: #2c2c2c;
}

.c0:focus,
.c0.focus {
  color: #222222;
  background: #ffffff;
  border-color: #2c2c2c;
}

.c0:disabled,
.c0.disabled {
  color: #222222;
  background: #ffffff;
  border-color: #222222;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > svg {
  float: left;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 > span {
  vertical-align: middle;
  white-space: nowrap;
}

<button
  class="c0 testBtn"
  data-testid="testProps"
  type="button"
  width="100px"
>
  <div
    class="c1"
  >
    <span>
      test Props
    </span>
  </div>
</button>
`;

exports[`test Button component should render the correct default button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  padding: 0 12px;
  height: 42px;
  font-family: 'PingFangSC-Light','sans-serif-light','Helvetica Neue For Number','-apple-system',BlinkMacSystemFont,'Segoe UI',Roboto,'Hiragino Sans GB','Microsoft YaHei','Helvetica Neue',Helvetica,Arial;
  font-size: 0.9375rem;
  line-height: 42px;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  width: 180px;
  color: #222222;
  background-color: #ffffff;
  border: 1px solid #222222;
}

.c0:hover {
  color: #222222;
  background: #ffffff;
  border-color: #2c2c2c;
}

.c0:focus,
.c0.focus {
  color: #222222;
  background: #ffffff;
  border-color: #2c2c2c;
}

.c0:disabled,
.c0.disabled {
  color: #222222;
  background: #ffffff;
  border-color: #222222;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > svg {
  float: left;
  width: 24px;
  height: 24px;
  margin-right: 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 > span {
  vertical-align: middle;
  white-space: nowrap;
}

<button
  class="c0"
  data-testid="default"
  type="button"
  width="180px"
>
  <div
    class="c1"
  >
    <span>
      Default
    </span>
  </div>
</button>
`;
